src/main/java/ArraySortedBag.java

```diff
@@ -212,7 +218,15 @@ public boolean equals(Object o) {
             return false;
         }
         ArraySortedBag<?> that = (ArraySortedBag<?>) o;
-        return Arrays.equals(this.bag, 0, this.rear, that.bag, 0, that.rear);
+        if (this.size() != that.size()) {
+            return false;
+        }
+        for (int i = 0; i < this.size(); i++) {
+            if (this.bag[i].compareTo((T) that.bag[i]) != 0) {
```

Comment:
This isn't really bag equality. Two bags are equal if they have the same elements appearing in the same cardinality. With this implementation, a sorted bag can never be equal to an unsorted bag -- the class check should only check if `that` is a `Bag`.

For example, see [Set#equals](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/AbstractSet.java#L85-L99) from the jdk.

Reply:
Seems like the easiest way to address this is to then just check if the `count` of each thing is the same between both bags. 

This change was made, but PR is marked as draft since the equals tests will need to be updated. 

Further, the same changes will need to be made for the indexed bag too.

Reply:
Still not quite right --- the hashcode now messes up due to ordering. 

I'll deal with this later.

Reply:
Nvm --- the strategy in the linked example (from abstract set) was to just remove the prime multiplier from the hashcode method.